protocols/identify/src/behaviour.rs

```diff
@@ -332,16 +364,18 @@ impl Behaviour {
 
     fn emit_new_external_addr_candidate_event(
         &mut self,
-        connection_id: ConnectionId,
+        _connection_id: ConnectionId,
         observed: &Multiaddr,
     ) {
-        if self
-            .outbound_connections_with_ephemeral_port
-            .contains(&connection_id)
-        {
-            // Apply address translation to the candidate address.
-            // For TCP without port-reuse, the observed address contains an ephemeral port which
-            // needs to be replaced by the port of a listen address.
+        if observed_port_matches_listening_port(observed, &self.listen_addresses) {
+            // If the observed port matches any of our listening ports,
+            // then port reuse actually worked. Use the original observed address.
+            self.events
+                .push_back(ToSwarm::NewExternalAddrCandidate(observed.clone()));
```

Comment:
```suggestion
        if observed_port_matches_listening_port(observed, &self.listen_addresses) 
            && ~self.external_addresses.as_slice().contains(observed)
        {
            // If the observed port matches any of our listening ports,
            // then port reuse actually worked. Use the original observed address.
            self.events
                .push_back(ToSwarm::NewExternalAddrCandidate(observed.clone()));
```

Otherwise we always emit the same addresses again.